lib/ansible/modules/network/iosxr/iosxr_logging.py

```diff
@@ -343,30 +697,29 @@ def main():
     argument_spec.update(element_spec)
     argument_spec.update(iosxr_argument_spec)
 
-    required_if = [('dest', 'hostnameprefix', ['name'])]
+    mutually_exclusive = [('dest', 'facility', 'hostnameprefix')]
+
+    required_if = [('dest', 'host', ['name']),
+                   ('dest', 'file', ['name']),
+                   ('dest', 'buffered', ['size', 'level']),
+                   ('dest', 'console', ['level']),
+                   ('dest', 'monitor', ['level'])]
```

Comment:
Please add mutually_exclusive and required_if in the aggregate spec as well. This ensures checks are applied to the individual item in aggregate.
```
    argument_spec = dict(
        aggregate=dict(type='list', elements='dict', options=aggregate_spec, mutually_exclusive= mutually_exclusive, required_if=required_if),
    )
```

Reply:
@ganeshrn you meant `self._module._check_required_if(required_if, item)` at line 223